airbyte-integrations/connectors/source-shortio/source_shortio/source.py

```diff
@@ -0,0 +1,266 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+import json
+import datetime
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.logger import AirbyteLogger
+
+class BasicAuthenticator(TokenAuthenticator):
+
+    def get_auth_header(self) -> Mapping[str, Any]:
+        if self.auth_method is None:
+            return {self.auth_header: f"{self._token}"}
+        else:
+            return {self.auth_header: f"{self.auth_method} {self._token}"}
+
+
+class Links(HttpStream, ABC):
+
+    url_base = "https://api.short.io/api"
+    limit = 150
+    primary_key = "id"
+    before_id = None
+    def next_page_token(
+        self, 
+        response: requests.Response
+        ) -> Optional[Mapping[str, Any]]:
+        
+        links = json.loads(response.text)['links']
+        try:
+            id_string = sorted(links, key=lambda k: k['createdAt'], reverse=False)[0]['idString']
+            if self.before_id != id_string:
+                self.before_id = id_string        
+                return id_string
+            else:
+                return None
+        except IndexError:
+            return None
+    def request_params(
```

Comment:
this formatting isn't consistent - can you make sure to run `./gradlew format` to make sure the code is formatted?